dashboard: add section with failover coordinator [WIP]

mandesero opened the pull request:
This patch introduces a new section displaying failover coordinator metrics. It includes panels for coordinator status (active/passive), and instances visible to each coordinator.

Need for #TNTP-197

Closes #247

If this pull request introduces support for new metrics, do not forget to
- [ ] add Grafana panel;
- [ ] run `make update-tests` to build new dashboard artifacts;
- [ ] add example application code to generate non-null metrics;
- [ ] attach a screenshot of a new panel to the PR description;
- [ ] add new Telegraf tag key, if relevant,
  - [ ] to all `telegraf.conf` files,
  - [ ] to "Grafana dashboard" documentation page;
- [ ] add alert example, if relevant,
  - [ ] to `alerts.yml`,
  - [ ] to "Alerting" documentation page;
- [ ] update `supported_metrics.md`.

In any case, do not forget to
- [ ] link an issue;
- [ ] add a CHANGELOG entry;
- [ ] update documentation guidelines;
- [ ] update README guidelines.

patapenka-alexey commented on dashboard/panels/cluster.libsonnet:
```diff
@@ -11,6 +11,192 @@ local prometheus = grafana.prometheus;
 
 {
   row:: common.row('Cluster overview'),
+  failover_coordinator_row:: common.row('Failover coordinator'),
```
I suggest to move this to the separate file `dashboard/panels/failover.libsonnet`, it will make all other changes cleaner.

patapenka-alexey commented on dashboard/section.libsonnet:
```diff
@@ -59,6 +59,12 @@ local vinyl = import 'dashboard/panels/vinyl.libsonnet';
+  failover_coordinator(cfg):: if cfg.type == variable.datasource_type.prometheus then [
+    cluster.failover_coordinator_row,
+    cluster.coordinators_status(cfg) { gridPos: { w: 10, h: 14, x: 0, y: 3 } },
+    cluster.instances_seen_by_coordinators(cfg) { gridPos: { w: 14, h: 14, x: 10, y: 3 } },
```
```suggestion
    failover.failover_coordinator_row,
    failover.coordinators_status(cfg) { gridPos: { w: 10, h: 14, x: 0, y: 3 } },
    failover.instances_seen_by_coordinators(cfg) { gridPos: { w: 14, h: 14, x: 10, y: 3 } },
```

patapenka-alexey commented on doc/monitoring/images/failover_coordinator_section.png:
Please, insert updated picture.

mandesero committed:
dashboard: add section with failover coordinator

This patch introduces a new section displaying failover coordinator
metrics. It includes panels for coordinator status (active/passive),
and instances visible to each coordinator.

Need for #TNTP-197

Closes tarantool#247

mandesero commented:
<img width="2824" height="1302" alt="image" src="https://github.com/user-attachments/assets/be002ca8-a36e-481d-9292-e22f71eb4a38" />

patapenka-alexey reviewed (approved):
Thank you for the patch!
